### DISCUSSION TITLE
How to interrupt long run shell commands?

### DISCUSSION THREAD

**Discussion by Yevgnen** [helix-editor/helix#3421] [Q&A]
Hi,

When I accidentally run a wrong long run shell command, helix is frozen and I have to kill it. How can I interrupt a running shell command? Thanks.

**Comment by the-mikedavis** [comment]
There is not currently a way. There are several rough edges for `:run-shell-command` and IMO it's a little out of scope to design around them since the shell command is mostly a stop-gap so that you can run at least _some_ external commands and keybind them. In the long-run there may be an integrated terminal that would be more appropriate for non-trivial shell commands (https://github.com/helix-editor/helix/issues/1976) and until then I would recommend using a separate shell (another terminal window, tmux split/tab/etc) for running non-trivial commands like `cargo build`, `cargo test`.

This particular case would be improved by https://github.com/helix-editor/helix/pull/3029; it would prevent helix from blocking on the command but you would still not have the ability to kill the command except in a separate shell with `kill`.

See also https://github.com/helix-editor/helix/issues/3337, https://github.com/helix-editor/helix/issues/3155

**Reply by Yevgnen** [reply]
I agree with using a separate shell for long run tasks. But sometimes I just accidentally type a wrong command like `find` or `tree` in the `~` (perhaps I forget that currently I'm at `~`) which lacks of changes to interrupt it...

**Reply by karimknaebel** [reply]
you can use `kill` in another shell to terminate the spawned process. not ideal, I know, but that's all we have for now
